src/numpy.jl

```diff
@@ -64,8 +64,8 @@ function npyinitialize()
     numpy = pyimport("numpy")
 
     # emit a warning if both Julia and NumPy are linked with MKL (#433)
-    if LinearAlgebra.BLAS.vendor() === :mkl &&
-       LinearAlgebra.BLAS.BlasInt === Int64 && hasproperty(numpy, "__config__")
+    julia_mkl = any(contains("mkl"), getfield.(BLAS.get_config().loaded_libs, :libname))
```

Comment:
`BLAS.get_config()` requires Julia v1.7, so I'd do something like
```suggestion
    julia_mkl = @static if VERSION < v"1.7-DEV"
        LinearAlgebra.BLAS.vendor() === :mkl
    else
        any(contains("mkl"), getfield.(BLAS.get_config().loaded_libs, :libname))
    end
```

Reply:
I removed `-DEV` since warning is better than not working